libs/mpi/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+find_package(MPI)
+
+if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
+    message(ERROR "cmake 3.12 or newer required")
+endif()
+
+# All cmake version don't have the same found variable set
+if(NOT MPI_FOUND AND NOT MPI_C_FOUND)
+  hpx_error("MPI could not be found and hpx::mpi library was requested\n
+  MPI_ROOT to point to the root of your MPI installation")
+endif()
+
+if (NOT TARGET hpx::mpi )
+  add_library(hpx::mpi INTERFACE IMPORTED)
+  set_property(TARGET hpx::mpi PROPERTY
+    INTERFACE_INCLUDE_DIRECTORIES ${MPI_INCLUDE_PATH} ${MPI_C_INCLUDE_DIRS})
+
+  if (MPI_C_LIBRARIES)
+      target_link_libraries(hpx::mpi INTERFACE ${MPI_CXX_LIBRARY})
+  endif()
+  # Ensure compatibility with older versions
+  if (MPI_LIBRARY)
+    target_link_libraries(hpx::mpi INTERFACE ${MPI_LIBRARY})
+  endif()
+  if (MPI_EXTRA_LIBRARY)
+    target_link_libraries(hpx::mpi INTERFACE ${MPI_EXTRA_LIBRARY})
+  endif()
+
+  set_property(TARGET hpx::mpi PROPERTY INTERFACE_COMPILE_OPTIONS ${MPI_CXX_COMPILE_FLAGS})
+    set_property(TARGET hpx::mpi PROPERTY INTERFACE_COMPILE_DEFINITIONS ${MPI_CXX_COMPILE_DEFINITIONS})
+endif()
```

Comment:
I would suggest to not re-discover MPI here but rely on the MPI logic that is already in place. Simply do not add this module to the mix if `hpx::mpi` is not a target.

Reply:
I will move all the logic to a common location. currently if networking is off, the find mpi stuff doesn't get called, so I duplicated it here. I want to fix it as it's implemented incorrectly anyway, it uses MPI_CXX_LIBRARY and includes instead of C_LIBRARY and includes etc etc